game.py

```diff
@@ -0,0 +1,148 @@
+            if self.code_number.isdigit():
+                self.code_number = int(self.code_number)
+                break
+            else:
+                print("Please enter a valid number.\n")
+        system("clear")
+        input('\nPress "Enter" to continue to the guesser\'s turn: ')
+        system("clear")
+        
+
+    def take_guess(self):
+        '''The decoder takes their guess in this function'''
+        # input() is used to pause the game before it continues to the next segment here. The prompt is in board.render
+        if not self.skip:
+            if self.guess_value > 0:
+                input()
+            print(f"\t\t\t\t\t\t\t\tTeam {self.team}.")
+            print(f"\t\t\t\t\t\t\tThe code word is --{self.code_word.capitalize()}--\n\t\t\t\t\t\t\t\tfor --{self.code_number}--\n")
+
+            flag = False
+            # while loop is used to check if the guess exists on the board and if it has already been guessed or not.
+            while not flag:
+                self.guess = input("Enter the word on the card you think this applies to: ").capitalize()
+                if self.guess in [card.word for card in self.board.cards if card.flipped]:
+                    print("This has already been guessed. Guess again.\n")
+                    continue
+                for card in self.board.cards:
+                    if self.guess == card.word:
+                        flag = True
+                        break
+                if not flag:
+                    print("Please enter a word on one of the cards. Guess again\n")
+                if flag:
+                    break
+            system("clear")
+        else:
+            input()
+    
+    def check_guess(self):
+        '''There are 4 options.
+        1)It is their teams word. 2)It is the other team's word 
+        3)It is a nuetral word 4)It is the death card and the game is over.'''
+        for card in self.board.cards:
+            if self.guess == card.word:
+                card.flipped = True
+                if self.team == card.color:
+                    self.guess_value = 1
+                elif self.team != card.color and card.color != "yellow" and card.color != "black":
+                    self.guess_value = 2
+                elif card.color == "yellow":
+                    self.guess_value = 3
+                elif card.color == "black":
+                    self.guess_value = 4
+                    self.game_over = True
+                break
+        if self.guess_value == 1:
+            self.code_number -= 1
+        else:
+            self.code_number = 0
+
+    def check_win(self):
+        '''Function ends the game on a few conditions:
+        1) If Blue or Red pick the death card the game will end.
+        2) If either team gets their entire word set the game will end.'''
+        color = ["Blue","Red"]
+
+        if self.game_over and not self.skip:
```

Comment:
It would be cleaner for each state update type function to "own" a set of game parameters, meaning they are the only function to set them. `self.game_over` is set in `check_guess()` and in `check_win()`. I'd recommend deleting `self.game_over = True` from `check_guess()` and moving it to `check_win()`. Maybe something like this...

```python
def check_win(self):
    ...

    if self.guess_value == 4:
        self.game_over = True
        print("Death card!")
```

This leaves "ownership" looking something like this...

`check_guess()` owns setting: `card.flipped`, `self.guess_value` (mostly), and `self.code_number`
`check_win()` owns setting: `self.game_over`
`update_state()` owns setting: `self.guessing`, `self.skip`, and `self.team`

Having this clear ownership will make your unit testing easier in the next step.